cephalopod/factory.py

```diff
@@ -68,3 +62,17 @@ def __call__(self, *args, **kwargs):
 
     celery.Task = ContextTask
     return celery
+
+
+@multipass.identity_handler
+def identity_handler(identity_info):
+    """Implements the Flask-Multipass identity handler"""
+    cephalopod_id = identity_info.data['cephalopod_id']
+    provider = identity_info.provider.name
+    whitelist = current_app.config['USER_WHITELIST']
+    if provider in whitelist and cephalopod_id in whitelist[provider]:
+        session['user'] = cephalopod_id
+        flash('You were successfully logged in', 'success')
+    else:
+        multipass.logout(url_for('.login'), clear_session=True)
```

Comment:
You are supposed to return the result from `multipass.logout` as it may redirect you to another location.